smap_l2_gridder/crs.py

```diff
@@ -0,0 +1,259 @@
+"""Info particular to geolocating the input data to a target grid."""
+
+from dataclasses import dataclass
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+from pyproj.crs import CRS
+from xarray import DataArray
+
+from .exceptions import InvalidGPDError
+
+
+@dataclass
+class Geotransform:
+    """Class for holding a GDAL-style 6-element geotransform."""
+
+    top_left_x: np.float64
+    pixel_width: np.float64
+    row_rotation: np.float64
+    top_left_y: np.float64
+    column_rotation: np.float64
+    pixel_height: np.float64
+
+    def col_row_to_xy(self, col: int, row: int) -> tuple[np.float64, np.float64]:
+        """Convert grid cell location to x,y coordinate."""
+        # Geotransform is defined from upper left corner as (0,0), so Adjust
+        # input value to the center of grid at (.5, .5)
+        adj_col = col + 0.5
+        adj_row = row + 0.5
```

Comment:
Oof, it's a GDAL style pixel, (but Y is the vertical and X is the horizontal.)

But think of this as the location of column and row, with the top left as the 0th column and 0th row.  The center of the pixel is the value that we want to store/know.

But the geotransform is defined at 0,0 the upper left edge of the upper left pixel. so this is just an adjustment /shorthand so that a user can ask for the  x-y location of the [0,0] grid cell and not have to adjust for the size of the gridcell.  

This is a quick tutorial on geotransforms.
https://gdal.org/en/stable/tutorials/geotransforms_tut.html

Reply:
Oh cool! So like an excel sheet. I'll go check out the geotransforms tutorial in my free time, seems like it'd be good to know.